Source/Core/Core/MemTools.cpp

```diff
@@ -60,11 +62,28 @@ static LONG NTAPI Handler(PEXCEPTION_POINTERS pPtrs)
     // virtual address of the inaccessible data
     uintptr_t fault_address = (uintptr_t)pPtrs->ExceptionRecord->ExceptionInformation[1];
     SContext* ctx = pPtrs->ContextRecord;
-
-    if (Core::System::GetInstance().GetJitInterface().HandleFault(fault_address, ctx))
+    Core::System& system = Core::System::GetInstance();
+    Memory::MemoryManager& memory = system.GetMemory();
+    if (system.GetJitInterface().HandleFault(fault_address, ctx))
     {
       return EXCEPTION_CONTINUE_EXECUTION;
     }
+    else if (!memory.IsPageDirty(fault_address))
```

Comment:
There's no check for whether the fault address is in emulated memory. When `IsPageDirty` runs its line `m_dirty_pages[index.value()]` for an address that isn't in emulated memory, it automatically adds an entry in `m_dirty_pages` for that index, which I assume isn't intentional.